nengo_loihi/hardware/allocators.py

```diff
@@ -84,11 +87,18 @@ def core_stdp_pre_cfgs(core):
     return profiles, profile_idxs
 
 
-def one_to_one_allocator(model):
-    board = Board()
-    chip = board.new_chip()
+class AbstractAllocator:
+    """Responsible for allocating the board's devices to models."""
+
+    def __call__(self, model):
+        """Returns a Board object corresponding to the given model."""
+        raise NotImplementedError()
+
+
+class OneToOne(AbstractAllocator):
+    """Assigns each block and input to distinct cores on the same chip."""
 
-    for block in model.blocks:
+    def block_to_chip(self, block, chip):
```

Comment:
Did GitHub's differ improve or something? I remember it butchering situations like #L101-132 before.